services/QuillLMS/spec/factories/teacher_notification.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: teacher_notifications
+#
+#  id            :bigint           not null, primary key
+#  email_sent    :datetime
+#  message_attrs :jsonb
+#  type          :text
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  user_id       :bigint           not null
+#
+# Indexes
+#
+#  index_teacher_notifications_on_user_id  (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+FactoryBot.define do
+  factory :teacher_notification do
+    user
+
```

Comment:
If you think `student_name` and `classroom_name` will be apart of all teacher notifications, I would just pull them up into the base factory.

Reply:
At this point, I just don't know.  I don't have a clear sense of what the next five or six notification types might be.  I could imagine doing some that only have classroom_name (like "your whole class has done X").  I do imagine that any time we have student_name we'll probably want to use classroom_name for disambiguation.